src/TestEditor.tsx

```diff
@@ -22,18 +27,25 @@ export default function TestEditor(props: Props): ReactElement {
     <div className="test-editor">
       <div className="test-editor-bar">
         <button
-          className="test-editor-run"
+          className={`test-editor-run ${props.runState?.status ?? ''}`}
           title="Run test"
-          onClick={() => {
-            /* TODO: Implement test run logic */
-          }}
+          onClick={() => props.onTestRun(props.test.testing_scope)}
+          disabled={props.runState?.status === 'running'}
         >
-          <span className="codicon codicon-play"></span>
+          <span
+            className={`codicon ${props.runState?.status === 'running' ? 'codicon-loading codicon-modifier-spin' : 'codicon-play'}`}
+          ></span>
         </button>
         <span className="test-editor-scope">
           <b>{props.test.testing_scope}</b> ➛{' '}
           {String(props.test.tested_scope.name)}
         </span>
+        {props.runState?.status === 'success' && (
+          <span className="test-run-success">Passed</span>
+        )}
+        {props.runState?.status === 'error' && (
+          <span className="test-run-error">Failed</span>
```

Comment:
For the future: the "failed" should store the Catala error message, to be able to parse it and determine what exactly went wrong (and to display it on the interface).